include/libp2p/common/final_action.hpp

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <functional>
```

Comment:
Copy code from `gsl::finally`/`gsl::final_action`.
- `F` instead of `std::function`
- movable (cancels moved)

Reply:
Added movable version of FinalAction